src/classes/OPP_AutomatedSoftCreditsService.cls

```diff
@@ -177,22 +177,27 @@ public class OPP_AutomatedSoftCreditsService {
             return accountToAffiliations;
         }
 
+        List<String> relatedContactRoles = new List<String>();
+        Schema.DescribeFieldResult fieldResult = npe5__Affiliation__c.Related_Opportunity_Contact_Role__c.getDescribe();
+        List<Schema.PicklistEntry> picklistValues = fieldResult.getPicklistValues();
+        for (Schema.PicklistEntry picklistValue : picklistValues) {
+            relatedContactRoles.add(picklistValue.getLabel());
+        }
+
         List<npe5__Affiliation__c> allAffiliations
             = new List<npe5__Affiliation__c>([SELECT Id, Name, npe5__Organization__c, npe5__Contact__c, Related_Opportunity_Contact_Role__c
                                                 FROM npe5__Affiliation__c
                                                 WHERE npe5__Organization__c IN :accountIds
+                                                    AND Related_Opportunity_Contact_Role__c IN :relatedContactRoles
                                                 ORDER BY CreatedDate DESC]);
 
         // Process the records that have a value in the Related_Opportunity_Contact_Role__c field
         for (npe5__Affiliation__c eachAffiliation : allAffiliations) {
-            if (eachAffiliation.Related_Opportunity_Contact_Role__c != null
-                && !eachAffiliation.Related_Opportunity_Contact_Role__c.containsIgnoreCase(RELATED_OPPORTUNITY_CONTACT_ROLE_NONE)) {
```

Comment:
This is interesting code. It's checking for the role not equal to "--none--". I would never expect that to be the value of the field. It's certainly not a picklist value of the field that we package. I can't really come up with any good reason this code would have been here. What do you think?

Reply:
Yea that was why I removed it, --none-- is a label in select list only and good catch on the other comment, I will change that to Value.